browse/services/documents/format_codes.py

```diff
@@ -69,67 +68,66 @@ def formats_from_source_flag(source_flag: str,
     F - PDF only
         PDF-only submission with .tar.gz package (likely because of anc files)
     """
-    formats = []
-    if not source_flag:
-        source_flag = ''
-    if not format_pref:
-        format_pref = ''
+    if isinstance(source_flag, SourceFlag):
+        source_flag = source_flag.code
```

Comment:
I wasn't going to comment on Union[str, SourceFlag], but this is pretty ugly. Note that the type of source_flag changes here to : str !

Reply:
Ya It would be nice to do this differently. Maybe get rid of the class `SourceFlag` and have the properties on `VersionEntry`?

Reply:
The `SourceFlag` is just a string with some properties to decode its meanings.